src/plugins/intel_cpu/src/transformations/snippets/x64/pass/mul_add_to_fma.cpp

```diff
@@ -27,7 +27,9 @@ ov::intel_cpu::pass::MulAddToFMA::MulAddToFMA() {
         const auto multiply = pattern_map.at(mul_m).get_node_shared_ptr();
         const auto add = pattern_map.at(add_m).get_node_shared_ptr();
 
-        if (transformation_callback(add)) {
+        if (transformation_callback(add) ||
+            std::dynamic_pointer_cast<snippets::modifier::MemoryAccess>(add) ||
+            std::dynamic_pointer_cast<snippets::modifier::MemoryAccess>(multiply)) {
```

Comment:
I think it should be in the matcher as you did in `BrgemmToBrgemmCPU`.
Because In general callbacks get supported ops and work with them. If we can reduce a count of nodes that are supported by the pass before callback call, we should do that! 😃

Reply:
Are you sure that it would be faster in this case?
When we match against a single node, I agree that we need to make this check as early as possible.
But when we match against a more complicated pattern, we might actually spend more time on dynamic_casts that would be redundant because there would be no required nodes (Add in this case).

@v-Golubev, you have more experience with the matchers, what do you think?
Has anyone profiled this kind of overheads?

Reply:
> Has anyone profiled this kind of overheads?

I have no relevant information about that. However, we always try to follow the principle that Alexandra perfectly described :)

I have another concern about that though. Do I understand correctly that we use `MemoryAccess` modifier meaning by this the nodes that were converted into TPP opset? If yes, maybe it is better to check TPP modifier in this case? This is CPU transformation, we already have an access to TPP headers here